### ISSUE TITLE
Each step will print log twice

### ISSUE THREAD

**Issue by BeihaiTianming** [ByteDance-Seed/VeOmni#223]
Epoch 1/5:   0%|          | 0/154 [00:07<?, ?it/s, loss: 2.66, grad_norm: 5.22, lr: 2.00e-05]
Epoch 1/5:   1%|          | 1/154 [00:07<20:13,  7.93s/it, loss: 2.66, grad_norm: 5.22, lr: 2.00e-05]
Epoch 1/5:   1%|          | 1/154 [00:21<20:13,  7.93s/it, loss: 2.59, grad_norm: 2.67, lr: 4.00e-05]
Epoch 1/5:   1%|▏         | 2/154 [00:21<28:01, 11.06s/it, loss: 2.59, grad_norm: 2.67, lr: 4.00e-05]
Epoch 1/5:   1%|▏         | 2/154 [00:34<28:01, 11.06s/it, loss: 2.58, grad_norm: 1.64, lr: 6.00e-05]
Epoch 1/5:   2%|▏         | 3/154 [00:34<30:00, 11.92s/it, loss: 2.58, grad_norm: 1.64, lr: 6.00e-05]
Epoch 1/5:   2%|▏         | 3/154 [00:47<30:00, 11.92s/it, loss: 2.24, grad_norm: 3.22, lr: 8.00e-05]

both GPU and NPU. And from the trend, the loss seems not decreased.

Epoch 5/5:  99%|█████████▊| 152/154 [35:22<00:27, 13.97s/it, loss: 2.47, grad_norm: 0.54, lr: 1.00e-04]
Epoch 5/5:  99%|█████████▉| 153/154 [35:22<00:13, 13.84s/it, loss: 2.47, grad_norm: 0.54, lr: 1.00e-04]
Epoch 5/5:  99%|█████████▉| 153/154 [35:36<00:13, 13.84s/it, loss: 2.34, grad_norm: 0.57, lr: 1.00e-04]
Epoch 5/5: 100%|██████████| 154/154 [35:36<00:00, 13.92s/it, loss: 2.34, grad_norm: 0.57, lr: 1.00e-04]
Epoch 5/5: 100%|██████████| 154/154 [35:36<00:00, 13.87s/it, loss: 2.34, grad_norm: 0.57, lr: 1.00e-04]

Here are my config
`model:
  model_path: Qwen3-32B
  attn_implementation: flash_attention_2

data:
  train_path: ./fineweb/sample/10BT/
  train_size: 100000000
  dataloader_type: native
  datasets_type: iterable
  data_type: plaintext
  max_seq_len: 32768
  text_keys: text
  drop_last: true

train:
  output_dir: XXX/Qwen3-32B
  data_parallel_mode: fsdp2
  ulysses_parallel_size: 8
  global_batch_size: 8
  micro_batch_size: 1
  rmpad: false
  rmpad_with_pos_ids: true
  bsz_warmup_ratio: 0.007
  dyn_bsz_margin: 0
  dyn_bsz_buffer_size: 200
  optimizer: adamw
  lr: 3.0e-4
  lr_warmup_ratio: 0.007
  lr_decay_style: constant
  lr_decay_ratio: 1.0
  weight_decay: 0.01
  max_grad_norm: 1.0
  enable_mixed_precision: true #false
  enable_gradient_checkpointing: true
  enable_full_shard: true
  enable_fsdp_offload: false
  enable_activation_offload: false
  init_device: meta
  enable_full_determinism: false
  empty_cache_steps: 200
  ckpt_manager: dcp
  load_checkpoint_path: ""
  save_steps: 1000
  num_train_epochs: 5
  save_hf_weights: false
  use_wandb: false`

**Comment by ZCMax** [comment]
@BeihaiTianming hello, I've encountered the same problem, have you fixed it?

**Comment by pjgao** [comment]
[pr290](https://github.com/ByteDance-Seed/VeOmni/pull/290) has fixed this ,  plz feedback any problems

**Comment by Coach257** [comment]
#290
